### PR TITLE
[SUSYBSMAnalysis/HSCP] Solve warning found by gcc 5.3.0

### PR DESCRIPTION
**Pull request by ghost**
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc530/CMSSW_8_1_X_2016-03-02-1100/SUSYBSMAnalysis/HSCP
The behaviour is unchanged, the parentheses makes clear the intended check while suppressing the warning.

### CODE REVIEW

**Comment by degano** [comment]
please test

**Comment by davidlt** [comment]
This again should be `==` to `!=`.

**Comment by davidlt** [comment]
No, this one might be fine, but I would prefer them not to use implicit conversion to bool. Someone with domain knowledge needs to look at this.

**Comment by davidlange6** [comment]
@degano - very belated feedback - I suspect the real intended meaning of the code is

anydifferentzero &= !(point->bx==0);
anydifferentone &= !(point->bx)==1);

could you implement that? Meanwhile @quertenmont may also comment

File: SUSYBSMAnalysis/HSCP/src/BetaCalculatorRPC.cc [hunk 1]
```diff
@@ -24,8 +24,8 @@ void BetaCalculatorRPC::algo(const std::vector<susybsm::RPCHit4D>& uHSCPRPCRecHi
   for(std::vector<susybsm::RPCHit4D>::iterator point = HSCPRPCRecHits.begin(); point < HSCPRPCRecHits.end(); ++point) {
     outOfTime |= (point->bx!=0); //condition 1: at least one measurement must have BX!=0
     increasing &= (point->bx>=lastbx); //condition 2: BX must be increase when going inside-out.
-    anydifferentzero &= (!point->bx==0); //to check one knee withoutzeros
-    anydifferentone &= (!point->bx==1); //to check one knee withoutones
+    anydifferentzero &= ((!point->bx)==0); //to check one knee withoutzeros
```
Comment (quertenmont) [inline SUSYBSMAnalysis/HSCP/src/BetaCalculatorRPC.cc]: I'd say it should be "point->bx!=0"

File: SUSYBSMAnalysis/HSCP/src/BetaCalculatorRPC.cc [hunk 2]
```diff
@@ -24,8 +24,8 @@ void BetaCalculatorRPC::algo(const std::vector<susybsm::RPCHit4D>& uHSCPRPCRecHi
...
+    anydifferentone &= ((!point->bx)==1); //to check one knee withoutones
```
Comment (quertenmont) [inline SUSYBSMAnalysis/HSCP/src/BetaCalculatorRPC.cc]: I'd say it should be "point->bx!=1"

**Comment by monttj** [comment]
Should we expect further implementation with "!="?
It looks fine as it is, though. 

Taejeong

**Pull request closed without merging**